test/Microsoft.IdentityModel.Tokens.Extensions.Tests/KeyVaultKeyWrapProviderTests.cs

```diff
@@ -0,0 +1,152 @@
+{
+    public class KeyVaultKeyWrapProviderTests
+    {
+        private readonly IKeyVaultClient _client;
+        private readonly SecurityKey _key;
+
+        public KeyVaultKeyWrapProviderTests()
+        {
+            _client = new MockKeyVaultClient();
+            _key = new KeyVaultSecurityKey(KeyVaultUtilities.CreateKeyIdentifier(), keySize: default);
+        }
+
+        [Theory, MemberData(nameof(DisposeProviderTheoryData))]
+        public void DisposeProviderTest(KeyWrapProviderTheoryData theoryData)
+        {
+            var context = TestUtilities.WriteHeader($"{this}.DisposeProviderTest", theoryData);
+
+            try
+            {
+                var provider = new KeyVaultKeyWrapProvider(_key, theoryData.Algorithm, _client);
+                _key.CryptoProviderFactory.ReleaseKeyWrapProvider(provider);
+
+                theoryData.ExpectedException.ProcessNoException(context);
+            }
+            catch (Exception exception)
+            {
+                theoryData.ExpectedException.ProcessException(exception, context);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<KeyWrapProviderTheoryData> DisposeProviderTheoryData
+        {
+            get => new TheoryData<KeyWrapProviderTheoryData>
+            {
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = SecurityAlgorithms.RsaPKCS1,
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    First = true,
+                    TestId = nameof(SecurityAlgorithms.RsaPKCS1),
+                },
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = SecurityAlgorithms.RsaOAEP,
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = nameof(SecurityAlgorithms.RsaOAEP),
+                },
+            };
+        }
+
+        [Theory, MemberData(nameof(KeyWrapProviderTheoryData))]
+        public void WrapUnwrapKeyTest(KeyWrapProviderTheoryData theoryData)
+        {
+            var context = TestUtilities.WriteHeader($"{this}.WrapUnwrapKeyTest", theoryData);
+
+            try
+            {
+                var provider = new KeyVaultKeyWrapProvider(_key, theoryData.Algorithm, _client);
+                Assert.NotNull(provider);
+
+                var keyBytes = Guid.NewGuid().ToByteArray();
+                var wrappedKey = provider.WrapKey(keyBytes);
+                Assert.NotNull(wrappedKey);
+                Assert.Equal(128, wrappedKey.Length);
+                Assert.False(Utility.AreEqual(keyBytes, wrappedKey), "keyBytes == wrappedKey");
+
+                var unwrappedKey = provider.UnwrapKey(wrappedKey);
+                Assert.NotNull(unwrappedKey);
+                Assert.True(Utility.AreEqual(keyBytes, unwrappedKey), "keyBytes != unwrappedKey");
```

Comment:
Consider using _IdentityComparer_ (_AreBytesEqual()_ for this specific case) so you can utilize CompareContext.
There are some other places where you could utilize the context.

We just put out a [Wiki page](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/How-We-Test) w.r.t. testing

Reply:
Fixed.